File: src/main/java/ar/edu/unlam/tallerweb1/domain/model/SharedExpent.java
package ar.edu.unlam.tallerweb1.domain.model;

import java.time.LocalDateTime;

public class SharedExpent {

    private Long id;
    private User owner;
    private Double amount;
    private String detail;
    private LocalDateTime date;
    private FriendsGroup friendsGroup;

    public SharedExpent(Long id, User owner, Double amount, String detail, LocalDateTime date, FriendsGroup friendsGroup) {
        this(owner, amount, detail, date, friendsGroup);
        this.id = id;
        this.friendsGroup = friendsGroup;
    }

    public SharedExpent(User owner, Double amount, String detail, LocalDateTime date, FriendsGroup friendsGroup) {
        this.owner = owner;
        this.amount = amount;
        this.detail = detail;
        this.date = date;
        this.friendsGroup = friendsGroup;
    }

    public Long getId() {
        return id;
    }

    public User getOwner() {
        return owner;
    }

    public Double getAmount() {
        return amount;
    }

    public String getDetail() {
        return detail;
    }

    public LocalDateTime getDate() {
        return date;
    }

    public FriendsGroup getFriendsGroup() {
        return friendsGroup;
    }

    public String getPayer() {
        return owner.getName();
    }
}

File: src/main/java/ar/edu/unlam/tallerweb1/domain/model/SharedExpent.kt
package ar.edu.unlam.tallerweb1.domain.model

import java.time.LocalDateTime

class SharedExpent(
    val owner: User,
    val amount: Double,
    val detail: String?,
    val date: LocalDateTime,
    var friendsGroup: FriendsGroup?
) {
    var id: Long? = null
        private set

    constructor(
        id: Long?,
        owner: User,
        amount: Double,
        detail: String?,
        date: LocalDateTime,
        friendsGroup: FriendsGroup?
    ) : this(owner, amount, detail, date, friendsGroup) {
        this.id = id
        this.friendsGroup = friendsGroup
    }

    val payer: String
        get() = owner.name
}
